src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -58,13 +60,37 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
         for charge in expunger.acquittals:
             charge.set_time_ineligible('Recommend sequential expungement', eligibility_date)
 
+    @staticmethod
+    def _calculate_has_subsequent_charge(class_b_felony, charges):
+        other_charges = [charge for charge in charges if charge != class_b_felony]
+        for other_charge in other_charges:
+            if other_charge.acquitted():
+                date_of_other_charge = other_charge.date
+            else:
+                date_of_other_charge = other_charge.disposition.date
+
+            if date_of_other_charge > class_b_felony.disposition.date:
+                return True
+        return False
+
     @staticmethod
     def _evaluate_class_b_felonies(expunger):
-        if expunger.most_recent_charge and expunger.most_recent_charge.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
-            for charge in expunger.class_b_felonies:
-                charge.set_time_ineligible('Time-ineligible under 137.225(5)(a)(A)(i)',
-                                           expunger.most_recent_charge.disposition.date + relativedelta(
-                                               years=TimeAnalyzer.TWENTY_YEARS))
+        for class_b_felony in expunger.class_b_felonies:
+            # If the class B felony is acquitted, then we have already handled its time eligibility.
+            # If it's convicted, this restriction overrides any of those rules.
+            if not class_b_felony.acquitted():
+                if TimeAnalyzer._calculate_has_subsequent_charge(class_b_felony, expunger.charges):
+                    # TODO: Express this with TimeEligibility with no eligibility date.
+                    type_eligibility = TypeEligibility(
+                        EligibilityStatus.INELIGIBLE,
+                        reason='137.225(5)(a)(A)(ii) - Class B felony can have no subsequent arrests or convictions')
+                    class_b_felony.expungement_result = ExpungementResult(type_eligibility=type_eligibility, time_eligibility=None)
+
+                else:
+                    if class_b_felony.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
+                        class_b_felony.set_time_ineligible('137.225(5)(a)(A)(i) - Twenty years from class B felony conviction',
+                                                       class_b_felony.disposition.date + relativedelta(
+                                                           years=TimeAnalyzer.TWENTY_YEARS))
```

Comment:
It seems like this is missing an else statement. If it doesn't have subsequent charges and the disposition date is less than 20 years what do we do? Otherwise perhaps we can do `elif` here.

Reply:
Yea, gotta fix this.

The comparison if `class_b_felony.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO` is checking that the date was after the date of 20 years ago, not that the conviction was "more than 20 years" old. The logic is correct by relying on the fact that if the felony is more than 20 years old and it has no subsequent charges, then it has already been set as eligible by the other parts of the time analyzer. 

I can rewrite this logic because it's difficult to interpret. The TimeAnalyzer can just have the constant `TWENTY_YEARS` and not `TWENTY_YEARS_AGO`, in keeping with the other time analysis implementation